### PR TITLE
Feature/async lock

### PR DESCRIPTION
**Pull request by mpluecker**
As this is dark magic, please review this carefully.

### CODE REVIEW

File: src/main/java/de/braintags/io/vertx/util/lock/AsyncReadWriteLock.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  }
+
+  private long obtainLock(boolean readLock) {
+    long lockStamp;
+    if (readLock) {
+      lockStamp = lock.tryReadLock();
+    } else {
+      lockStamp = lock.tryWriteLock();
+    }
+    return lockStamp;
```
Comment (kerko) [inline src/main/java/de/braintags/io/vertx/util/lock/AsyncReadWriteLock.java]: Das geht auch viel kürzer, aber geschmackssache:
`retrun readLock ? lock.tryReadLock() : lock.tryWriteLock()`

File: src/main/java/de/braintags/io/vertx/util/asyncLock/AsyncLock.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package de.braintags.io.vertx.util.asyncLock;
```
Comment (remi128) [inline src/main/java/de/braintags/io/vertx/util/asyncLock/AsyncLock.java]: add header for copyright and license, check code templates from eclipse

File: src/main/java/de/braintags/io/vertx/util/asyncLock/AsyncLock.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package de.braintags.io.vertx.util.asyncLock;
+
+public class AsyncLock {
```
Comment (remi128) [inline src/main/java/de/braintags/io/vertx/util/asyncLock/AsyncLock.java]: add documentation

File: src/test/java/de/braintags/io/vertx/util/asyncLock/AsyncReadWriteLockTest.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
+package de.braintags.io.vertx.util.asyncLock;
```
Comment (remi128) [inline src/test/java/de/braintags/io/vertx/util/asyncLock/AsyncReadWriteLockTest.java]: Please add the test to Suite TestAllUtil

File: src/test/java/de/braintags/io/vertx/util/asyncLock/AsyncReadWriteLockTest.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+          handler.run();
+          if (simulateIllegalHandler) {
+            handler.run();
+          }
+        };
+        if (simulateException) {
+          throw new RuntimeException("test execption in worker");
+        }
+
+        es.submit(runnable);
```
Comment (remi128) [inline src/test/java/de/braintags/io/vertx/util/asyncLock/AsyncReadWriteLockTest.java]: The future is not used anymore? Does the use of this future clash with the vertx future?

File: src/main/java/de/braintags/io/vertx/util/asyncLock/AsyncReadWriteLock.java [hunk 6]
```diff
@@ -0,0 +1,145 @@
+package de.braintags.io.vertx.util.asyncLock;
```
Comment (remi128) [inline src/main/java/de/braintags/io/vertx/util/asyncLock/AsyncReadWriteLock.java]: In general: really dark magic :)
Please open discussion under vertx-developer or vertx to check potential side affects

**Pull request merged**